api/tests/views_tests.py

```diff
@@ -34,34 +60,136 @@ def test_runtime_data(client):
     assert response.status_code == 200
 
 
-@pytest.mark.django_db
-def test_free_mask_email_limit_error(settings) -> None:
-    """A JSON error is returned when a free user hits the mask limit"""
-    free_user = make_free_test_user()
-    free_profile = free_user.profile_set.get()
+def test_post_domainaddress_success(prem_api_client) -> None:
+    """A premium user can create a domain address."""
+    response = prem_api_client.post(
+        reverse("domainaddress-list"),
+        data={"address": "my-new-mask"},
+        format="json",
+    )
+    assert response.status_code == 201
+    ret_data = response.json()
+    assert ret_data["enabled"]
+    assert ret_data["full_address"].startswith("my-new-mask@premium.")
+
+
+def test_post_domainaddress_no_subdomain_error(prem_api_client) -> None:
+    """A premium user needs to select a subdomain before creating a domain address."""
+    premium_profile = get_user(prem_api_client).profile_set.get()
+    premium_profile.subdomain = ""
+    premium_profile.save()
+
+    response = prem_api_client.post(
+        reverse("domainaddress-list"),
+        data={"address": "my-new-mask"},
+        format="json",
+    )
+
+    assert response.status_code == 403
+    ret_data = response.json()
+    assert ret_data == {
+        "detail": (
+            "You must select a subdomain before creating email address with subdomain."
+        ),
+        "error_code": "need_subdomain",
+    }
+
+
+def test_post_domainaddress_user_flagged_error(prem_api_client) -> None:
+    """A flagged user cannot create a new domain address."""
+    premium_profile = get_user(prem_api_client).profile_set.get()
+    premium_profile.last_account_flagged = timezone.now()
+    premium_profile.save()
+
+    response = prem_api_client.post(
+        reverse("domainaddress-list"),
+        data={"address": "my-new-mask"},
+        format="json",
+    )
+
+    assert response.status_code == 403
+    ret_data = response.json()
+    assert ret_data == {
+        "detail": "Your account is on pause.",
+        "error_code": "account_is_paused",
+    }
+
+
+def test_post_domainaddress_bad_address_error(prem_api_client) -> None:
+    """A domain address can be rejected due to the address format or content."""
+    response = prem_api_client.post(
+        reverse("domainaddress-list"),
+        data={"address": "myNewAlias"},
+        format="json",
+    )
+
+    assert response.status_code == 403
```

Comment:
suggestion (non-blocking): this one seems like it should be `400` rather than `403`. I.e., no matter what `Authorization` or authentication is sent with this request, it will fail because of the content in the request - not because of the access level of the user.